index.js

```diff
@@ -1,255 +1,32 @@
-        , body: JSON.stringify({
-            e: 'ls'
-            , ls: payload
-        }), auth: {
-            username: config.key
-        }, headers: {
-            'Content-Type': 'application/json; charset=UTF-8'
-            , 'user-agent': config.UserAgent
-        }, timeout: MAX_REQUEST_TIMEOUT_MS
-        , withCredentials: false
-        , agent: new agent.HttpsAgent({
-            freeSocketTimeout: FREE_SOCKET_TIMEOUT_MS
-        })
-    };
-
-    // Flush the Log
-    async.retry({
-        times: MAX_REQUEST_RETRIES
-        , interval: retryCount => REQUEST_RETRY_INTERVAL_MS * Math.pow(2, retryCount)
-        , errorFilter: errCode => DEFAULT_HTTP_ERRORS.includes(errCode) || errCode === 'INTERNAL_SERVER_ERROR'
-    }, (reqCallback) => {
-        return request(options, (error, response, body) => {
-            if (error) return reqCallback(error.code);
-            if (response.statusCode >= INTERNAL_SERVER_ERROR) return reqCallback('INTERNAL_SERVER_ERROR');
-            return reqCallback(null, body);
-        });
-    }, callback);
-};
-
-// Main Handler
-exports.handler = (event, context, callback) => {
-    const config = getConfig(event)
-        , eventData = processEvent(parseEvent(event))
-        , payload = config.eventlog ? [prepareEvent(eventData)] : []
-        , s3params = {
-            Bucket: eventData && eventData.meta && eventData.meta.bucket && eventData.meta.bucket.name || undefined
-            , Key: eventData && eventData.meta && eventData.meta.object && eventData.meta.object.key || undefined
-        };
-
-    if (config.filelog) {
-        return getLogs(s3params, (error, lines) => {
-            if (error) {
-                if (config.eventlog) return send(payload, config, callback);
-                return callback(error);
-            }
-
-            return async.everySeries(batchify(payload.concat(prepareLogs(lines, eventData))), (batch, next) => {
-                setTimeout(() => {
-                    return send(batch, config, next);
-                }, BATCH_INTERVAL_MS);
-            }, callback);
-        });
-    } else if (!config.eventlog) {
-        return callback('None of file and event logging has been enabled!');
-    }
-
-    return send(payload, config, callback);
-};
+'use strict'
+
+const {once} = require('events')
+
+const {handleEvent} = require('./lib/event-handler.js')
+const {createLoggerClient} = require('./lib/logger.js')
+const {getLogs, prepareLogs} = require('./lib/transformer.js')
+const {getProperty} = require('./lib/utils.js')
+
+module.exports = {
+  handler
+}
+
+async function handler(event, context) {
+  const eventData = handleEvent(event)
+  const s3params = {
+    Bucket: getProperty(eventData, 'meta.bucket.name')
+  , Key: getProperty(eventData, 'meta.object.key')
+  }
+
+  const logger = createLoggerClient(s3params.Bucket)
+  const lines = await getLogs(s3params)
+  logger.on('error', console.error)
+  const logs = prepareLogs(lines, eventData)
```

Comment:
`prepareLogs` can return undefined if `lines` is invalid, which will then throw on line 25. should `prepareLogs` always return an array instead?

Reply:
Great idea!